### PR TITLE
Implementation for DenseElementsAttr -> DenseResourceElementsAttr

### LINKED ISSUE #137: Switch to using `DenseResourceElementsAttr`

### PR DESCRIPTION
**Pull request by saienduri**
Add implementation to use DenseResourceElementsAttr instead of DenseElementsAttr.

Fixes #137

### CODE REVIEW

File: python/shark_turbine/dynamo/backends/cpu.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def _base_backend(gm: torch.fx.GraphModule, example_inputs):
     importer.import_graph_module(gm)
     print(module, file=sys.stderr)
     with context:
-        pm = PassManager.parse("builtin.module(torch-to-iree)")
+        pm = PassManager.parse("builtin.module(torch-to-iree,iree-util-import-resources)")
```
Comment (stellaraccident) [inline python/shark_turbine/dynamo/backends/cpu.py:76]: I don't think you should need this? (probably just was for testing?)

File: python/shark_turbine/dynamo/importer.py [hunk 2]
```diff
@@ -94,6 +100,24 @@
     torch.complex128: "complex<f64>",
 }
 
+TORCH_DTYPE_TO_IREE_TYPE: Dict[torch.dtype, Callable[[], MlirType]] = {
```
Comment (stellaraccident) [inline python/shark_turbine/dynamo/importer.py:103]: We're going to be upstreaming the importer, so let's not refer to IREE. Change this to TORCH_DTYPE_TO_MLIR_TYPE

File: python/shark_turbine/dynamo/importer.py [hunk 3]
```diff
@@ -889,6 +913,16 @@ def _make_constant_op(
     )
 
 
+def create_iree_tensor_type(tensor: torch.Tensor) -> MlirType:
```
Comment (stellaraccident) [inline python/shark_turbine/dynamo/importer.py:916]: Also rename this to create_mlir_tensor_type

File: python/shark_turbine/dynamo/importer.py [hunk 4]
```diff
@@ -94,6 +100,24 @@
...
+    torch.bfloat16: lambda: BF16Type.get(),
+    torch.float32: lambda: F32Type.get(),
+    torch.float64: lambda: F64Type.get(),
+    torch.uint8: lambda: IntegerType.get_unsigned(8),
+    torch.int8: lambda: IntegerType.get_signed(8),
+    torch.int16: lambda: IntegerType.get_signed(16),
+    torch.int32: lambda: IntegerType.get_signed(32),
+    torch.int64: lambda: IntegerType.get_signed(64),
+    torch.bool: lambda: IntegerType.get_signless(1),
+    torch.qint8: lambda: IntegerType.get_signless(8),
```
Comment (stellaraccident) [inline python/shark_turbine/dynamo/importer.py:114]: This should be get_signed.

File: python/shark_turbine/dynamo/importer.py [hunk 5]
```diff
@@ -94,6 +100,24 @@
...
+    torch.quint8: lambda: IntegerType.get_signless(8),
```
Comment (stellaraccident) [inline python/shark_turbine/dynamo/importer.py:115]: And this get_unsigned.

**Review by stellaraccident — CHANGES_REQUESTED** [review]

File: python/shark_turbine/dynamo/backends/cpu.py (hunk 1 above)
Comment (saienduri) [reply python/shark_turbine/dynamo/backends/cpu.py]: Yeah, this is not needed. Thanks for spotting that.

**Comment by stellaraccident** [comment]
Sai, are the pre-requisites met to rebase/merge this?

**Comment by saienduri** [comment]
Yeah, should I bump iree in turbine first before we land this or either way is fine?

**Review by stellaraccident — APPROVED** [review]

**Pull request merged**